### PR TITLE
CLI tools: Use target project when we cannot find startup project

### LINKED ISSUE #11330: CLI Tools: Default unfound startup project to target project
Two scenarios:
* You run `dotnet ef` in a directory with multiple project files
* You run `dotnet ef` in a directory without projects

In both of these scenarios, we'll fail telling you to specify `--project`, but once you do that, we'll fail again telling you to also specify `--startup-project`.

It would be nice if when we can't find the startup project, but you specified a target project we'd just use the target project. (and vice-versa if you specify a startup project but we can't find a target project)

### PR DESCRIPTION
**Pull request by smitpatel**
and vice-versa

Resolves #11330

cc: @bricelam - how to test this works?

### CODE REVIEW

File: src/dotnet-ef/RootCommand.cs [hunk 1]
```diff
@@ -204,41 +199,97 @@ protected override int Execute()
...
+                        throw new CommandException(Resources.MultipleProjectsInDirectory(startupProjectPath));
+                    }
+
+                    return (projects[0], startupProjects[0]);
+                }
+                else
+                {
+                    return startupProjects.Count == 1
+                        ? (projects[0], startupProjects[0])
+                        : (projects[0], projects[0]);
```
Comment (bricelam) [inline src/dotnet-ef/RootCommand.cs]: Missing case: If there are multiple startup projects in the current directory, it should throw.

File: src/dotnet-ef/RootCommand.cs [hunk 2]
```diff
@@ -204,41 +199,97 @@ protected override int Execute()
...
+                    }
+
+                    if (startupProjects.Count != 1)
+                    {
+                        throw new CommandException(Resources.MultipleProjectsInDirectory(startupProjectPath));
+                    }
+
+                    return projects.Count == 1
+                        ? (projects[0], startupProjects[0])
+                        : (startupProjects[0], startupProjects[0]);
```
Comment (bricelam) [inline src/dotnet-ef/RootCommand.cs]: Ditto. (throw when multiple)

File: src/dotnet-ef/RootCommand.cs (hunk 1 above)
Comment (bricelam) [reply src/dotnet-ef/RootCommand.cs]: Only use `project` when there are no startup projects.

**Comment by bricelam** [comment]
Manually test the various cases. We had automated tests, but they were flaky and a bit hacky.

**Comment by smitpatel** [comment]
@bricelam - Updated PR to work/throw/use the other project appropriately.

Test manually on the metric of 36 difference cases through truth table.

**Pull request merged**